src/assets/PythonCopyFileWithShutil/python-copy-paste.md

```diff
@@ -0,0 +1,105 @@
+
+### 1. shutil.copy() method
+
+```py
+shutil.copy(source, destination, *, follow_symlinks=True)
+```
+
+This method copies the file of source to the directory or file specified in the destination and returns file's destination.
+
+- shutil.copy() method copies file's data and file's permission mode.
+- Metadata(file creation and modification time) of the file copied is not preserved.
+- Source and destination should be path-like objects or strings.
+- Destination can specify both file or directory. If the destination specify path to a directory, the file will be copied to destination directory with the basename of the source path.
+- Copying symbolic link:
+
+  - If `follow_symlinks=True`, destination will be a copy of the file that the symbolic link in source points to. Default value of parameter `follow_symlinks` is `True`.
+
+  - If `follow_symlinks=False`, destination will be created as symbolic link.
+
+  ```py
+  # Import shell utility module
+  import shutil
+
+  #  Copy the content of source_file.txt to destination_file.txt
+  shutil.copy('source_file.txt', 'destination_file.txt')
+  ```
+
+### 2. shutil.copy2() method
+
+```py
+shutil.copy2(source, destination, *, follow_symlinks=True)
+```
+
+The shutil.copy2() method provides additional functionality of preserving all the metadata of the file it can. Other characteristics are same as shutil.copy() method.
+  - shutil.copy2() method copies file's data and file's permission mode.
+  - Source and destination should be path-like objects or strings.
+  - Destination can specify both file or directory. If the destination is directory, the file will be copied with the same name as of the source file name.
+
+    ```py
+    # Import shell utility module
+    import shutil
+
+    #  Copy the content of source_file.txt to destination_file.txt
+    shutil.copy2('source_file.txt', 'destination_file.txt')
+    ```
+
+### 3. shutil.copyfile() method
+
+```py
+shutil.copyfile(source, destination, *, follow_symlinks=True)
+```
+Using this method, a file's content is copied from its source to its destination without any metadata.
+
+  - Source and destination should be path-like objects or strings.
+  - Destination path can only specify path to a file but cannot specify path to a directory.
+
+    ```py
+    # Import shell utility module
+    import shutil
+
+    #  Copy the content of source_file.txt to destination_file.txt
+    shutil.copyfile('source_file.txt', 'destination_file.txt')
+    ```
+
+### 4. shutil.copyfileobj() method
+
+```py
+shutil.copyfileobj(fsrc, fdst, length)
+```
+
+- Source and destination must be file-like objects.
+- This method copies contents of file object to the destination file object.
+- Length parameter which is optional, specifies an integer value for buffer size.
```

Comment:
```suggestion
- Length parameter, which is optional, specifies an integer value for buffer size.
```

This seems strange, why is the parameter called "length" but controls "buffer size"? Is that just "the way it is" in python?